### DISCUSSION TITLE
How to Skip Frames in the Stream

### DISCUSSION THREAD

**Discussion by vaghawan** [miguelaeh/pipeless#152] [Q&A]
Hi, 
skipping frames is an essential feature, is there any option to do that while submitting the stream to the pipeless? I can't see in the comman line option, maybe I'm missing something. 

Thanks
V

**Comment by miguelaeh** [comment]
Hi @vaghawan ,
Yes, that feature is implemented. You can set the `--stream-buffer-size` indicating the max number of frames to queue. If your processing gets slow, then once that number of frames is queued it will discard the next ones until there is space.

**Reply by vaghawan** [reply]
Thank you @miguelaeh , what is the default value for this param?

**Reply by vaghawan** [reply]
Hi, I've an odd issue, when I run the yolo and object tracking stages, it runs fine, the output is fine. But once I run a stage which may take a little bit more time for the attribute detection, my output file is corrupted and no longer plays. The file is also very low, This is how I'm running it, 
`pipeless add stream --input-uri file:////media1/samples/cam3.mp4  --output-uri file:///home/test-pipe/my-project/output-solider.mp4 --frame-path "yolo,object-tracking,asoilder" `

The stages are fine, I've tested them, but collectively something off, right now, I'm just trying to write the original frames as my final output without modification, even then, it doesn't work. There is no apparent error , the detection are printed in the console, everything needed to be run are running fine. It's quite odd at this point. Any help would be appreciated.

**Reply by miguelaeh** [reply]
I would bet the problem in your case is a known bug lossing the EOS packet from the input. Likely due to the stream-buffer-size being too small.
This should not happen, but we detected the bug in the past. Mp4 files require a final special packet called moov, and if the EOS is lost, the file never finished the creation. You can set the output to screen and you should see the stream properly

**Reply by vaghawan** [reply]
Ohh, is there any other way, I do not have a display attached to the device, this is usually not the case of our application. It happens almost everytime. Is it same, if I forward the output as rtsp? 

or maybe, I can save flv, if that works? 

Thanks

**Reply by vaghawan** [reply]
I think, this is the bug you're referring to is what is happening in my case as well, I tried another video, now the pipeline closes smoothly, 
```
[2024-08-16T12:02:49Z INFO  pipeless_ai::pipeline] End of input stream reached. Pipeline id: cfc5c70c-4ff5-43c9-be1d-3f7a03bc9430
[2024-08-16T12:02:50Z INFO  pipeless_ai::output::pipeline] End of stream in output gst pipeline. Pipeline id: cfc5c70c-4ff5-43c9-be1d-3f7a03bc9430
[2024-08-16T12:02:50Z INFO  pipeless_ai::pipeline] End of output stream reached for pipeline: cfc5c70c-4ff5-43c9-be1d-3f7a03bc9430
[2024-08-16T12:02:50Z INFO  pipeless_ai::events] Stream loop stopped
[2024-08-16T12:02:50Z INFO  pipeless_ai::dispatcher] Stream config entry removed. Stopping associated pipeline
```

However, when I get the corrupt output, what I get is just this, 

```
[2024-08-16T11:14:44Z INFO  pipeless_ai::input::pipeline] Received received EOS from source pipeline0.
                    Pipeline id: ee2f845d-8a4a-4e7e-a17c-5255e4cf6531 ended

```

**Reply by miguelaeh** [reply]
That message indicates the EOS is read from the input, but since the message is discarded when there is not enough capacity,  the EOS does not reach the output causing the problem. You can increase the buffer size using the `--stream-buffer-size` flag

**Reply by vaghawan** [reply]
I see, let me try it, thanks, I was doing exactly opposite reducing the --stream-buffer-size . 

Thanks
